R/Filter.R

```diff
@@ -266,18 +266,14 @@ randomForestSRC.rfsrc = makeFilter(
 #' @name makeFilter
 NULL
 
-# for some reason we cannot call 'randomForestSRC_var.select' directly as we then face
-# nested recursion problems when using other methods than "md".
 rf.min.depth = makeFilter(
   name = "randomForestSRC_var.select",
   desc = "Minimal depth of / variable hunting via method var.select on random forests fitted in package 'randomForestSRC'.",
   pkg = "randomForestSRC",
   supported.tasks = c("classif", "regr", "surv"),
   supported.features = c("numerics", "factors", "ordered"),
   fun = function(task, nselect, method = "md", ...) {
-    im = randomForestSRC::var.select(getTaskFormula(task), getTaskData(task),
-      method = method, verbose = FALSE, ...)$md.obj$order
-    setNames(-im[, 1L], rownames(im))
+    # redirected to randomForestSRC.var.select()
```

Comment:
Wait, I'm missing something here. How does this redirect work?

Reply:
Because down the line, `randomForestSRC.var.select` does another `.FilterRegister[["randomForestSRC.var.select"]] = randomForestSRC.var.select` call.

There can only one function be registered. I found this when debugging; we always jump to the body of the function who last called `.FilterRegister` for a unique name.

So in detail this is what happens

1. `rf.min.depth` registers `rf.min.depth`, calls it "deprecated" and then registers the filter under the name `randomForestSRC_var.select`.

2. `randomForestSRC.var.select` does the same as 1) and hence the body of this filter will finally be used when someone calls `randomForestSRC_var.select` as it overwrites 1)

Yes, this is "deprecation hell" but I am not eager on touching more in here.
So currently we have 2 deprecated methods (`randomForestSRC.var.select` and `rf.min.depth`) and the current one `randomForestSRC_var.select` which all call the same function body

Reply:
Ok, thanks.